prin

```diff
@@ -0,0 +1,149 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Set PROGram name
+PROG=${0##*/}
+########################################################################
+#+
+#+ NAME
+#+     $PROG - What tag(s) is my PR IN?
+#+
+#+ SYNOPSIS
+#+     $PROG  <commit|pr>
+#+     $PROG  [--security_layer=/path/to/pointer/to/script]
+#+     $PROG  [--helpshort|--usage|-?]
+#+     $PROG  [--help|-man]
+#+
+#+ DESCRIPTION
+#+     Report the tags in which a PR or commit appears and the latency in
+#+     days (commit to release).
+#+
+#+ OPTIONS
+#+     [--security_layer=]       - A file containing a path to a script to
+#+                                 source:
+#+                                 FLAGS_security_layer=/path/to/script
+#+     [--help | -man]           - display man page for this script
+#+     [--usage | -?]            - display in-line usage
+#+
+#+ EXAMPLES
+#+
+#+ FILES
+#+
+#+ SEE ALSO
+#+     common.sh                 - base function definitions
+#+
+#+ BUGS/TODO/IDEAS
+#+     Limit results to first tag/release per branch
+#+
+########################################################################
+# If NO ARGUMENTS should return *usage*, uncomment the following line:
+usage=${1:-yes}
+
+source $(dirname $(readlink -ne $BASH_SOURCE))/lib/common.sh
+source $TOOL_LIB_PATH/gitlib.sh
+
+###############################################################################
+# FUNCTIONS
+###############################################################################
+
+#############################################################################
+# Display/Report tags and latency (to release) for the incoming commits
+# K8S latency is the difference between the commit timestamp and tag timestamp
+# @param pr - a PR or "" (placeholder)
+# @param commits - a list of commits
+# HOSTED_EPOCH is set by an external script
+show_tags () {
+  local pr=$1
+  local initial_commit=$2
+  shift 2
+  local cp_commits=($*)
+  local commit
+  local commit_seconds
+  local tag_seconds
+  local tag
+  local days
+  local hosted_days
+
+  logecho
+  logecho "COMMITS: $initial_commit (initial) ${cp_commits[*]} (cherrypicks)"
+  [[ -n $pr ]] && echo "(https://github.com/kubernetes/kubernetes/pull/$pr)"
+  logecho
+  logecho "                LATENCY (DAYS)"
+  logecho -n "TAG             K8S"
+  if ((SECURITY_LAYER)); then
+    logecho "     HOSTED"
+  else
+    logecho
+  fi
+  logecho -n "---             ---"
+  if ((SECURITY_LAYER)); then
+    logecho "     ------"
+  else
+    logecho
+  fi
+
+  # We only care about the initial commit for latency calculation
+  commit_seconds=$(git show -s --format="%ct" $initial_commit)
+
+  for commit in $initial_commit ${cp_commits[*]}; do
+    for tag in $(git tag --contains $commit); do
```

Comment:
Yes, that or instead of assuming a workspace, create a temporary (or cached) one.  I'm not sure if the convenience is worth the extra overhead